### DISCUSSION TITLE
how to know user and display in reconnectwm.sh?

### DISCUSSION THREAD

**Discussion by Kreijstal** [neutrinolabs/xrdp#2978] [Q&A] [answered]
I want to execute the following
```
sudo -u $MYUSER DISPLAY=$MYDISPLAY DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/$MYUSERID/bus cinnamon-screensaver-command --exit
```

I do not want to type the lock password

**Comment by matt335672** [comment, marked as answer]
The reconnectwm.sh script runs as the reconnecting user. The environment should be correct already.

Temporarily set up a reconnectwm.sh script containing the following:-

```sh
#!/bin/sh

exec >/tmp/reconnect.log 2>&1
env | sort
```

Then try a reconnect and see if you've got the environment set you need in /tmp/reconnect.log.

**Reply by Kreijstal** [reply]
If reconnectwm executes as the user how do you make different reconnect scripts for different users?

**Reply by metalefty** [reply]
I believe it can be done in any way. For example, you can switch the code path to reach depending on the user like this. Other whatever way will work. The code works as you write. Do it in your favourite way.

```sh
case "$USER" in
    alice|bob)
        exec /path/to/reconnect-script-alice-bob.sh
    ;;
    charlie)
        date >> /tmp/charlie.log
        printenv >> /tmp/charlie.log
        exit
    ;;
    *)
        exec /path/to/reconnect-script-default.sh
    ;;
esac
```

**Reply by Kreijstal** [reply]
understood, I had the false belief it checked something on your home configuration to override it, but actually this might be better because you can have choice.

**Reply by metalefty** [reply]
per-user startwm.sh ($HOME/startwm.sh) is supported but I remember the same thing for reconnect.sh is not supported  at the moment.
